crates/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -83,8 +88,16 @@ impl CrateDefMap {
 
         // First parse the root file.
         let root_file_id = context.crate_graph[crate_id].root_file_id;
+
+        #[cfg(not(feature = "aztec"))]
         let ast = parse_file(&mut context.file_manager, root_file_id, errors);
 
+        #[cfg(feature = "aztec")]
+        {
+            let mut ast = parse_file(&mut context.file_manager, root_file_id, errors);
+            aztec_contracts_macros(&mut ast);
+        }
```

Comment:
Does this still work if the `aztec` feature flag is set? I would have thought the block here prevents `ast` from being visible outside the block. The `ast` line is also the same as line 93, sans mutability. We could rewrite it to avoid that if `aztec_contracts_macros` moved ast into and out of itself:

```rs
        let ast = parse_file(&mut context.file_manager, root_file_id, errors);

        #[cfg(feature = "aztec")]
        let ast = aztec_contracts_macros(ast);
```